app/services/globus_digest_generator.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# GlobusDigestGenerator
+#
+# Add digests to Globus DROs if the files currently lack them. Existing files
+# will not be overwritten.
+class GlobusDigestGenerator
+  # @param [Cocina::Model] cocina a DRO to add digests to
+  # @param [Hash] globus_ids a mapping of filenames to their location on disk
+  # @param [Boolean] validate whether to build the new DRO using validation
+  # @return [Cocina::Model] a new Cocina object with (potentially) new digests
+  def self.generate(cocina:, globus_ids:, validate: true)
+    new(cocina: cocina, globus_ids: globus_ids, validate: validate).generate
+  end
+
+  def initialize(cocina:, globus_ids:, validate:)
+    @cocina = cocina
+    @globus_ids = globus_ids
+    @validate = validate
+  end
+
+  def generate
+    return cocina unless cocina.dro?
+
+    props = cocina.to_h
+    props[:structural] = gen_filesets(props[:structural])
+
+    if validate
+      Cocina::Models.build(props)
+    else
+      Cocina::Models.build_request(props)
+    end
+  end
+
+  private
+
+  attr_reader :cocina, :globus_ids, :validate
+
+  def gen_filesets(filesets)
```

Comment:
We typically avoid abbreviations like "gen". (But definitely not a hard rule or anything.)

Reply:
Makes sense -- I replaced with `generate`.